install/tools/ipa-ca-install

```diff
@@ -116,9 +116,22 @@ def parse_options():
     return safe_options, options, filename
 
 
-def get_dirman_password():
-    return installutils.read_password(
-        "Directory Manager (existing master)", confirm=False, validate=False)
+def _get_dirman_password(password=None, unattended=False):
+    if not password:
+        if unattended:
+            sys.exit('Directory Manager password required')
+        try:
+            password = installutils.read_password(
+                "Directory Manager (existing master)", confirm=False,
+                validate=False)
+        except KeyboardInterrupt:
+            sys.exit(0)
```

Comment:
I'm afraid that this is inconsistent with other parts of code. The rest will return error code 1 in case of SIGINT.

If you want to keep new error code, then manpage should be updated as well.